src/lib/components/StoragePlace.svelte

```diff
@@ -26,33 +36,66 @@
   class="rounded-sm border m-3 inline-block h-fit min-w-80 max-w-[420px] border-black p-1"
   role="tree"
 >
-  <h1 class="font-bold">{storagePlaceName} <span class="text-stone-400">({items.list.length})</span></h1>
-  <div role="group">
-    {#each items.list as item, i (item.id)}
-      <Item
-        bind:item={items.list[i]}
-        isSelected={items.selected === i}
-        onSelected={(amount = 0) => {
-          items.select(i + amount)
-        }}
-        onDelete={items.delete}
-      />
-    {/each}
-  </div>
-  {#if items.list.length === 0}
-    <div class="text-stone-400">Nothing in the {storagePlaceName}.</div>
+  <h1 class="font-bold">{storagePlaceName}
+    {#if items} <span class="text-stone-400">({items.list.length})</span> {/if}
+  </h1>
+  {#if items}
+    <div role="group">
+      {#each items.list as item, i (item.id)}
+        <Item
+          bind:item={items.list[i]}
+          isSelected={items.selected === i}
+          onSelected={(amount = 0) => {
+            if (items)
+              items.select(i + amount)
+          }}
+          onDelete={() => {
+            if (items)
+              items.delete(item.id)
+          }}
+          onUpdate={(updatedItem: StoredItem) => {
+            if (items)
+              items.update(item.id, updatedItem)
```

Comment:
wondering if u have an idea why typescript throws the error "`items` might be null" here, though this element should only appear if items exists... ?

Reply:
no error for me there
![image](https://github.com/ashlsun/untitled-inventory-app/assets/36823200/9e1f2893-5cee-4698-bb93-50a0c3e77e7e)

Reply:
oh without the `if (items)` line i mean - maybe typescript doesnt take into account the {#if items} from svelte?

Reply:
weird typescript issue, and this is a temp fix. i still haven't figured out how to best work with stores and a db.